Content.Shared/Posters/SharedPosterSystem.cs

```diff
@@ -0,0 +1,175 @@
+using Content.Shared.DoAfter;
+using Content.Shared.Hands.EntitySystems;
+using Content.Shared.Interaction;
+using Content.Shared.Posters.Components;
+using Content.Shared.Tag;
+using Content.Shared.Verbs;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Network;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Posters;
+
+public abstract partial class SharedPosterSystem : EntitySystem
+{
+    [Dependency] private readonly TagSystem _tagSystem = default!;
+    [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+    [Dependency] private readonly SharedHandsSystem _handsSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly INetManager _net = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<FoldedPosterComponent, AfterInteractEvent>(OnFoldedInteract);
+        SubscribeLocalEvent<FoldedPosterComponent, PosterPlacingDoAfterEvent>(OnPlacingDoAfter);
+        SubscribeLocalEvent<FoldedPosterComponent, GetVerbsEvent<UtilityVerb>>(OnPlacingVerb);
+
+        SubscribeLocalEvent<PosterComponent, GetVerbsEvent<AlternativeVerb>>(OnRemoveVerb);
+        SubscribeLocalEvent<PosterComponent, PosterRemovingDoAfterEvent>(OnRemovingDoAfter);
+    }
+
+    private void OnFoldedInteract(EntityUid uid, FoldedPosterComponent folded, AfterInteractEvent args)
+    {
+        if (folded.CancelToken != null || !args.CanReach || args.Target == null)
+            return;
+
+        if (!HasComp<TagComponent>(args.Target))
+            return;
+
+        if (folded.PosterPrototype == null)
+            return;
+
+        if (!_prototypeManager.HasIndex<EntityPrototype>(folded.PosterPrototype))
+            return;
+
+        if (_tagSystem.HasTag(args.Target.Value, "Wall"))
+            StartPlacing(uid, folded, args.Target.Value, args.User);
+    }
```

Comment:
This is way too hardcoded for posters, components should be as generic as possible. This should just be a wallplaceablecomponent instead.

Reply:
> This is way too hardcoded for posters, components should be as generic as possible. This should just be a wallplaceablecomponent instead.

I can't find this component, not sure it exists (maybe you mean Wallmount), but system still need to check if someone trying to apply poster on the wall and not on other object. I think, i'll just add Tag field in component to make component less hardcoded.